engine/runtime/src/test/java/org/enso/interpreter/test/ForeignMethodInvokeTest.java

```diff
@@ -3,20 +3,44 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
+import java.util.Random;
 import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 import org.graalvm.polyglot.Context;
 import org.graalvm.polyglot.Value;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
-import org.junit.Ignore;
 import org.junit.Test;
 
 public class ForeignMethodInvokeTest extends TestBase {
   private static Context ctx;
 
   @BeforeClass
   public static void prepareCtx() {
-    ctx = defaultContextBuilder("enso", "js").build();
+    var r = new Random();
+    var counter = new long[1];
+    var b =
+        defaultContextBuilder("enso", "js")
+            .onDeniedThreadAccess(
```

Comment:
With `onDeniedThreadAccess` as proposed by https://github.com/oracle/graal/pull/8266 we have a way to [act like an Ethernet on collision](https://github.com/enso-org/enso/pull/8885/commits/160f2b81617f459a1a2bf198f940c78e7e7db9ed#diff-a84ceb8a331a33ef55d986aac04b55c48c52f2daddf6bea30911823ab24ff3a7R24) and successfully orchestrate the two threads accessing the same context to run into completion.

More info on [Ethernet collision detection and behavior](https://en.wikipedia.org/wiki/Carrier-sense_multiple_access_with_collision_detection) - most important _"Calculate and wait the random backoff"_.